site/app/templates/grading/electronic/ta_status/StatusBase.twig

```diff
@@ -118,6 +119,23 @@
                 return mode;
             }
         {% endif %}
+        {% if component_averages|length != 0  %}
+            let bucketInputManualValue = parseFloat(bucketInputManual.value);
+            if(bucketInputManual.value!=''){
+                if (isNaN(bucketInputManualValue) || bucketInputManualValue === 0) {
+                    alert('Error: Invalid bucket size.');
+                    return ;
+                }
+                mode=fillBuckets(range,bucketInputManualValue,betterBuckets,xValue,yValue,xBuckets,mode,modeCount,max,min);
+                var dataAuto = tracing(xBuckets,yValue);
+                Plotly.newPlot('myDiv4', dataAuto, layout, {displayModeBar: false,displaylogo: false});
+                bucketInputManual.value='';
+                document.getElementById("mySmallDiv4").innerHTML='<b>Mode: </b>'
+                +mode+'&nbsp&nbsp&nbsp&nbsp'+'<b>Median: </b>'+median+'&nbsp&nbsp&nbsp&nbsp'+ '<b>Maximum: </b>'+max+
+                '&nbsp&nbsp&nbsp&nbsp'+'<b>Minimum: </b>'+min+'&nbsp&nbsp&nbsp&nbsp'+'<b>Range: </b>'+range;
```

Comment:
Here's another opportunity for an XSS attack.  It would  be better to pre-render the HTML structure and fill in only the relevant values via JS.